### ISSUE TITLE
Automatic Versioning Help

### ISSUE THREAD

**Issue by SeanCarrick** [PekinSOFT-Systems/Northwind-Basic#4]
I've tried to create a way for the application to automatically increment its version information. I do the math on the build number. Once the build reaches 9999, it should roll over to 0. The rest of the version numbers should calculate based off of the build number, automatically incrementing when the time is right.

However, the build number is not increasing, therefore, nothing else will automatically increment. Any ideas on what I messed up? Here's the code, which is in the static initializer (`static{}`) of the `Application` class:

```java
        // Pull in the application version information from the properties file. 
        long bui = Long.valueOf(props.getProperty("app.build", "0"));
        int rev = Integer.valueOf(props.getProperty("app.revision", "0"));
        int min = Integer.valueOf(props.getProperty("app.minor", "1"));
        int maj = Integer.valueOf(props.getProperty("app.major", "0"));
        
        log.debug("Calculating the version of the application.");
        if ( Boolean.getBoolean(props.getProperty("debugging")) ) {
/* These commented out lines were an attempt to keep the build number between 1583 (just a 
 * random number that I picked) and 9999. I commented them out when the automatic version 
 * incrementing was not working and I started trying to debug the reason. Now, I am just adding
 * one to the build number that is read into the system from the properties file and it's still not
 * working. Also, I was originally going to do some math with the system time in milliseconds for
 * the build number. However, I decided against that because it could work out that the rest of the
 * version numbers would not increment as expected.
 */
//            if ( bui == 0 ) {
//                bui = 1583;
//            } else {
//                System.out.println("Current System Time in Milliseconds: " +
//                        System.currentTimeMillis());
                bui++;
//            }

            // If the build hits 10000, reset it the build to zero (0) and increment the 
            //+ revision by one (1).
            if ( bui > 9999 ) {
                bui = 0;
                rev++;
            }
            // If the revision hits 101, reset the revision to zero (0) and increment the 
            //+ minor by one (1).
            if ( rev > 100 ) {
                rev = 0;
                min++;
            }
            // If the minor hits 11, reset the minor to zero (0) and increment the 
            //+ major by one (1).
            if ( min > 10 ) {
                min = 0;
                maj++;
            }
        }
        
        // Store the calculated values to the constants and back into the application 
        //+ properties.
        MAJOR = maj; props.setProperty("app.major", String.valueOf(MAJOR));
        MINOR = min; props.setProperty("app.minor", String.valueOf(MINOR));
        REVISION = rev; props.setProperty("app.revision", String.valueOf(REVISION));
        BUILD = bui; props.setProperty("app.build", String.valueOf(BUILD));
```

No matter how many times I look at this code, I cannot figure out where my mistake is. Please help me figure this out, as I believe that I might be getting close to running the project 1583 times and want it to start incrementing. Thank you for your assistance with this.

**Comment by jkovalsky** [comment]
In my opinion the if condition at line 151 is wrong and should have been this instead:

`if ( Boolean.parseBoolean(props.getProperty("debugging")) ) {`

Now the build number increments automatically if I set `debugging=true` in my ~/.northwind.conf file.

**Comment by SeanCarrick** [comment]
Updated that line and used the `DEBUGGING` global setting in the condition. Thanks for your assistance @jkovalsky 

Pushed and merged the changes with Pull #6 so I'm closing the issue.
